### ISSUE TITLE
include generate into helm install

### ISSUE THREAD

**Issue by rimusz** [helm/helm#388]
would be nice to have  an extra `helm install -g` flag to run `generate` first, than `install`
That would allow to run two commands with the one command instead of running two commands:
```
helm install
helm generate
```

**Comment by helgi** [comment]
`helm install` should scan the templates and run generates as needed IMO - No flag

**Comment by krancour** [comment]
I would agree with @helgi

**Comment by rimusz** [comment]
what about the case that only once some e.g. secrets need to be generated?

**Comment by helgi** [comment]
`install` != `update/upgrade` - Muddling that creates issues, even if it feels like it is convenient

Possibly a generate file could be flagged (by it self) as it should only run once for the lifecycle unless forced, similar to how apt-get can run `reconfigure` on package but otherwise it never reconfigures it

**Comment by rimusz** [comment]
@helgi you are right for the most cases.
Ok lets have install without the flag:
`helm install should scan the templates and run generates as needed IMO - No flag`

**Comment by rimusz** [comment]
`helm install -g` is implemented, so we can close this one
